javalin/src/main/java/io/javalin/http/staticfiles/StaticFileConfig.kt

```diff
@@ -2,12 +2,14 @@ package io.javalin.http.staticfiles
 
 import io.javalin.core.util.Header
 import org.eclipse.jetty.server.handler.ContextHandler.AliasCheck
+import java.util.function.Predicate
 
 data class StaticFileConfig(
     @JvmField var hostedPath: String = "/",
     @JvmField var directory: String = "/public",
     @JvmField var location: Location = Location.CLASSPATH,
     @JvmField var precompress: Boolean = false,
     @JvmField var aliasCheck: AliasCheck? = null,
-    @JvmField var headers: Map<String, String> = mutableMapOf(Header.CACHE_CONTROL to "max-age=0")
+    @JvmField var headers: Map<String, String> = mutableMapOf(Header.CACHE_CONTROL to "max-age=0"),
+    @JvmField var filter: Predicate<String> = Predicate { true },
```

Comment:
If we're keeping it like this, then I guess it should be called `pathFilter`, but we could also give it a "Context", so there is a bit more flexibility?

Reply:
Best would be to give it a Context, offers the most flexibility - for example, it could make decision based on request headers.
I wanted to do it this way initially, but the ConfigurableHandler.getResource() method has only path available, I'm not sure how to get the full Context there, that's way I went with Predicate of String instead of Predicate of Context, even though the Context would be better.

Reply:
You would have to change

```kotlin
interface ResourceHandler {
    fun handle(httpRequest: HttpServletRequest, httpResponse: HttpServletResponse): Boolean
    fun addStaticFileConfig(config: StaticFileConfig)
}
```

to

```kotlin
interface ResourceHandler {
    fun handle(ctx: Context): Boolean
    fun addStaticFileConfig(config: StaticFileConfig)
}
```

But it becomes a bigger change. If you want, I make a separate change which changes to use `Context` ?

Reply:
Yes if you could make a change in ResourceHandler.handle() method then i can easily implement the filter takie a Context.

For my private uses case, the Predicate of String is enough, but i think sooner or later someone will ask how to make a filter that can access request headers, so Predicate of Context is more future-proof.

Reply:
Switching to `Context` is difficult, since the handler takes a `JavalinResponseWrapper(res, rwc)`. This wrapper does compression, etags, etc. You can make the filter take `HttpServletRequest` instead ?

Reply:
Yes i think it should not depend on response object, only on request

Reply:
Great, then I think you should add the test to line 37, something like:

```kotlin
override fun handle(httpRequest: HttpServletRequest, httpResponse: HttpServletResponse): Boolean {
   if (config.isAllowed(req) == false) return false
```

Reply:
I added the change. Extracted the check to ConfigurableHandler's "canHandleRequest" method- this class is marked "open" so I thought this logic may be also overridable. But if you don't like it as a separate method, feel free to inline this check and remove the method.